src/ui/public/vis/vis_types/vislib_vis_type.js

```diff
@@ -99,13 +99,14 @@ export function VislibVisTypeProvider(Private, $rootScope, $timeout, $compile) {
     }
 
     destroy() {
+      $(this.container).find('vislib-legend').remove();
```

Comment:
sometimes we get in a weird state when legend doesn't get cleared. seems this.container no longer points to the right dom node. needs to be further investigated.